src/control/util.rs

```diff
@@ -30,6 +33,20 @@ impl<S> Backoff<S> {
             wait_dur,
         }
     }
+
+    fn poll_timer(&mut self) -> Poll<(), S::Error> {
```

Comment:
It looks like this never returns an error, so I wonder if this is really the appropriate return type( (instead of, say, `Async<()>`)? In that case, we wouldn't need the `S: Service` bound on this impl.

On the other hand, this lets us use `try_ready!` in the `Service` impl for `Backoff` below, so maybe it's the right thing. 🤷‍♀️